src/main.cpp

```diff
@@ -675,36 +676,33 @@ void GetGlobalStatus()
 
 std::string GetCurrentOverviewTabPoll()
 {
-    std::string poll = "";
-    std::string sMessageKey = ExtractXML(msPoll, "<MK>", "</MK>");
-    std::string sPollExpiration = ExtractXML(msPoll, "<EXPIRATION>", "</EXPIRATION>");
-    uint64_t uPollExpiration = 0;
-    // Alerts are displayed as polls but do not have an expiration
-    if(sPollExpiration.empty())
-    {
-        uPollExpiration = pindexBest->nTime;
+    AssertLockHeld(MinerStatus.lock);
+
+    // The global msPoll variable contains the poll most-recently published to
+    // the network. If it hasn't expired, return the title of this poll:
+    if (PollIsActive(msPoll)) {
+        return ExtractXML(msPoll, "<MK>", "</MK>").substr(0, 80);
     }
-    else
-    {
-        try
-        {
-            uPollExpiration = stoll(sPollExpiration);
-        }
-        catch(std::exception &e)
-        {
-            // Malformed poll expiration, don't display
-            uPollExpiration = 0;
+
+    // Otherwise, find the most recent active poll from the AppCache:
+    AppCacheSection::value_type const* selected_ptr = nullptr;
```

Comment:
I think we should use a copy here. While `ReadCacheSection` returns a reference now, if we change that this will break.

Reply:
Done 🙂 I just went with a string for a faster copy, but `optional` is pretty neat. I figured the pointer was okay because we're talking about moving away from the AppCache, but good to be safe!